app/parse.py

```diff
@@ -15,8 +29,92 @@ class Product:
     num_of_reviews: int
 
 
-def get_all_products() -> None:
-    pass
+PRODUCT_FIELD = [field.name for field in fields(Product)]
+
+
+_driver: WebDriver | None
+
+
+def get_driver() -> WebDriver:
+    return _driver
+
+
+def set_driver(new_driver: WebDriver) -> None:
+    global _driver
+    _driver = new_driver
+
+
+def get_single_product(product_soup: Tag) -> Product:
+    return Product(
+        title=product_soup.select_one(".title")["title"],
+        description=product_soup.select_one(
+            ".description").text.replace("\xa0", " "),
+        price=float(product_soup.select_one(
+            ".price").text.replace("$", "")),
+        rating=len(product_soup.select(".glyphicon-star")),
+        num_of_reviews=int(product_soup.select_one(
+            ".ratings > p.pull-right").text.split()[0]
+                           ),
+    )
+
+
+def check_accept_cookies(driver: WebDriver) -> None:
+    try:
+        accept_cookies_button = driver.find_element(
+            By.CLASS_NAME, "acceptCookies"
+        )
+        if accept_cookies_button:
+            accept_cookies_button.click()
+            time.sleep(0.1)
+    except:
+        pass
+
+
+def check_more_pages(driver: WebDriver) -> None:
+    try:
+        more_button = driver.find_element(
+            By.CLASS_NAME, "ecomerce-items-scroll-more"
+        )
+        if more_button:
+            while more_button.value_of_css_property(
+                    "display") == "block":
+                more_button.click()
+                time.sleep(0.1)
+    except:
+        pass
```

Comment:
don't use "total except", write what error it should catch
write smth instead of pass""